src/components/[guild]/RoleCard/components/EditRewardDescription/EditRewardModalContent.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react"
+import useToast from "../../../../../hooks/useToast"
+import { Controller, FormProvider, useForm } from "react-hook-form"
+import useEditGuildPlatform from "../../../AccessHub/hooks/useEditGuildPlatform"
+import {
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  ModalBody,
+  ModalCloseButton,
+  ModalFooter,
+  ModalHeader,
+} from "@chakra-ui/react"
+import RichTextDescriptionEditor from "../../../RolePlatforms/components/AddRoleRewardModal/components/AddContractCallPanel/components/CreateNftForm/components/RichTextDescriptionEditor"
+import Button from "../../../../common/Button"
+import { GuildPlatform } from "../../../../../types"
+
+type Props = {
+  guildPlatform: GuildPlatform
+  onClose: () => void
+}
+
+export const EditRewardModalContent: React.FC<Props> = ({
+  guildPlatform: { id, platformGuildData },
+  onClose,
+}) => {
+  const methods = useForm({ defaultValues: { ...platformGuildData } })
+  const { formState, control, handleSubmit } = methods
+
+  const toast = useToast()
+
+  const { onSubmit, isLoading } = useEditGuildPlatform({
+    guildPlatformId: id,
+    onSuccess: () => {
+      toast({
+        status: "success",
+        title: "Successfully updated reward description",
+      })
+      onClose()
+    },
+  })
+
+  return (
+    <FormProvider {...methods}>
```

Comment:
I think we don't really need a provider here, since don't use the `useFormContext` hook inside the child components